swagger.yaml

```diff
@@ -0,0 +1,108 @@
+openapi: 3.0.3
+info:
+  title: Agartex Service
+  description: |-
+    This is a server for compiling Latex documents and authenticating users
+  version: 0.0.1
+servers:
+  - url: http://localhost:3000
+  - url: some_domain
+tags:
+  - name: user
+    description: Operations about user
+paths:
+  /users:
+    post:
+      tags:
+        - user
+      summary: Register user
+      description: This will register a new user with the given email and password.
+      operationId: createUser
+      requestBody:
+        description: Created user object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Credentials'
+      responses:
+        201:
+          description: Successfully created user
+        400:
+          description: Malformed request
+        409:
+          description: Duplicate email
+        422:
+          description: Request body validation errors (e.g. incorrect email format, weak password)
+          content:
+            text/plain:
+              schema:
+                type: string
+  /sessions:
+    post:
+      tags:
+        - user
+      summary: Logs user into the system
+      description: |-
+        The session ID is returned in a cookie called 'RSESSID' and it must be included in authentication.
+        Required options: Secure, HttpOnly
+      operationId: loginUser
+      requestBody:
+        description: Login Credentials
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Credentials'
+      responses:
+        201:
+          description: Successfully created session
+          headers:
+            Set-Cookie:
+              description: Session token
+              schema:
+                type: string
+                example: RSESSID=token_value; Secure; HttpOnly
+        400:
+          description: Malformed request body
+        401:
+          description: Authentication using supplied email and password failed
+        422:
+          description: Request body validation errors (e.g. incorrect email format, weak password)
+          content:
+            text/plain:
+              schema:
+                type: string
```

Comment:
Why do we need format validation on `/sessions` endpoint? I would keep only `201`, `400` and `401` response codes.